src/components/DateInputs/DateRangeInputs.tsx

```diff
@@ -40,12 +40,27 @@ const DateRangeInputs: React.FC<Props> = (props) => {
 
     const setFromValue = (fromValue: Dayjs | null): void => {
         setLocalFromValue(fromValue);
-        setRangeIfPossible(fromValue, localToValue);
+        if (
+            fromValue &&
+            localToValue &&
+            fromValue > localToValue &&
+            fromValue >= reasonableMinDate
+        ) {
```

Comment:
Can we extract this and the complicated boolean check below into little helper function functions (plain `.ts`, outside the component) with descriptive names so it's easy to read at a glance what the check is doing?